### DISCUSSION TITLE
Form submissions with optional fields

### DISCUSSION THREAD

**Discussion by ytsdv** [pocketbase/pocketbase#5483] [Q&A]
I am currently using pocketbase to save submissions from a simple form app.
When a user does not answer a optional numeric field (the field is not present in the payload to pocketbase) it gets set to 0 in the database entry. 
This is an issue because 0 could be a valid answer and i need to differentiate between skipped and answered fields.

After using pocketbase for 2 previous projects this is the first time i ran into this issue. 

A workaround seems to be to send a predefined magic constant instead, which i would rather not do.

Anyone implemented something similar with pocketbase?

**Comment by ganigeorgiev** [comment]
I'll consider documenting it better but this is an intended behavior and a very early implementation decision (see https://github.com/pocketbase/pocketbase/issues/122).

All PocketBase fields (with exception of the `json` type) are **non-nullable and uses zero-defaults** similar to Go (for type `number` that is 0, for `text` empty string, etc.). For all zero-defaults you can check the first example value listed in the https://pocketbase.io/docs/collections/#overview table.

In general you have 3 options:

- use a special distinct value (magic constant, as you've noted)
- use an additional `bool` field to indicate "skipped"
- use `json` field type instead of `number`

There are no other workarounds at the moment.
